libfulltext/arxiv/test_arxiv.py

```diff
@@ -0,0 +1,68 @@
+# copyright © 2018 the libfulltext authors (see AUTHORS.md and LICENSE)
+"""Unit tests for arXiv"""
+
+from unittest import TestCase
+import requests
+import hashlib
+
+from . import get_arxiv_fulltext
+
+
```

Comment:
Here I was wondering how we plan to handle multiple files from a single identifier, and added this drafted assert function. I guess the idea is clear, I'm not sure if the `checklist` handling actually works (I expect it's clear'ish what I had in mind), yet when running tests I found that the APS DOIs I wanted to test with, didn't work …

Reply:
I see what you had in mind. It feels a little indirect for me. And this applies to the resulting code on the calling side, lines 37-48, as well. What I wonder is:

- Why are the assertions for `True` in the checklists not made in the `assert_sha1_checklist` function?
- Would it maybe make sense to pass a list of dicts / named tuples with data for each paper rather than 3 separate lists?

Reply:
hm, apparently I didn't click submit on my reply:

 - I like the dict idea, implemented now.
 - the problem with the checklist is that in `assert_sha1_checklist` the checklist cannot be checked. That function only returns the `save_stream`-like function, so at `assert_sha1_checklist` runtime, not a single file has been downloaded. and in the returned function we only see one file at a time.